insights/v1alpha1/0000_10_01_datagather.crd.yaml

```
@@ -0,0 +1,242 @@
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  annotations:
+    api-approved.openshift.io: https://github.com/openshift/api/pull/1365
+    include.release.openshift.io/ibm-cloud-managed: "true"
+    include.release.openshift.io/self-managed-high-availability: "true"
+    include.release.openshift.io/single-node-developer: "true"
```

If this is a v1alpha1 it needs to be installed only as a techpreview
```suggestion
    include.release.openshift.io/single-node-developer: "true"
    release.openshift.io/feature-set: TechPreviewNoUpgrade
```
You'll also need to configure the schemapatch generator to generator tech preview manifests in this new API, create a `.codegen.yaml` file at the group level and copy the schemapatch configuration from the example, https://github.com/openshift/api/blob/94ac871f4a586d03b0ae11344f3604016065a458/example/.codegen.yaml#L7-L12 (you can drop the disabled field as that's the default value)

---

Ahh OK. Hmm could we go with `v1beta1` then? I mentioned my concern in the https://github.com/openshift/enhancements/pull/1279#discussion_r1037848835

---

Having this as a techpreview is a natural choice, and we are dropping the idea of beta APIs within openshift, you have a choice of alpha or GA.

My recommendation would be introduce this as tech preview, we are trying to introduce all new features as TP. We can either deploy separate deployment versions with a feature flag via cli using CVO annotations or you can detect the TP state within the operator itself. The former is probably easier for you.

---

OK TP it is. This should be updated. Thanks